lib/srv/desktop/tdp/proto_test.go

```diff
@@ -200,3 +188,137 @@ func TestMFA(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, respWant, respGot)
 }
+
+func TestIsNonFatalErr(t *testing.T) {
+	// Test that nil returns false
+	require.False(t, IsNonFatalErr(nil))
+	// Test that any other error returns false
+	require.False(t, IsNonFatalErr(errors.New("some other error")))
+}
+
+// TDP messages must have size limits in order to prevent attacks that
+// soak up system memory. At the same time, exceeding such size limits shouldn't
+// kill a user's running session, or else that becomes a DoS attack vector.
+// To this end, TestSizeLimitsAreNonFatal checks that exceeding size limits causes
+// only non-fatal errors.
+func TestSizeLimitsAreNonFatal(t *testing.T) {
+	for _, test := range []struct {
+		name string
+		msg  Message
+	}{
+		{
+			name: "rejects long ClientUsername",
```

Comment:
Shouldn't this case be fatal though? The length of a username is capped by Windows, so if this happens, it's not legitimate and there's no need for us to try and continue the session.

Reply:
Good point